tests/rptest/clients/rpk.py

```diff
@@ -901,7 +901,10 @@ def _execute(self, cmd, stdin=None, timeout=None, log_cmd=True):
             output, stderror = p.communicate(input=stdin, timeout=timeout)
         except subprocess.TimeoutExpired:
             p.kill()
-            raise RpkException(f"command {' '.join(cmd)} timed out")
+            output, stderr = p.communicate()
+            raise RpkException(
+                f"command {' '.join(cmd)} timed out, output: {output} \n error: {stderr}",
```

Comment:
Does this work in practice?  I would expect the first call to communicate may well have drained anything interesting from stdout & stderr, and indeed the second call may also time out.

Reply:
Yeah it does, if you check out some of the failed delete-records tests in the previous runs you can grep for `dbg` it was placed in this exception handler here, it does print the output. I thought this was strange too, but after taking a look at the pdocs for `communicate` i saw this is exactly how to handle timeouts
```
The child process is not killed if the timeout expires, so in order to cleanup properly a well-behaved application should kill the child process and finish communication:
```
https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

Reply:
I tried this out on a quick little python script + bash shell that prints out stuff.  Looks like the second `p.communicate()` will collect all of the output from the executed function